lib/autoproj/cli/main_ci.rb

```diff
@@ -15,11 +15,19 @@ class MainCI < StandaloneCI
             option :report, type: "string", default: "cache-pull.json",
                             desc: "a file which describes what has been pulled"
             def build(*args)
-                if (cache = options.delete(:cache))
-                    cache = File.expand_path(cache)
+                if (cache = options[:cache]) || File.exist?(options[:report])
                     require "autoproj/cli/base"
                     Autoproj::CLI::Base.validate_options(args, options)
-                    results = cache_pull(cache, ignore: options.delete(:cache_ignore))
+                    results = if cache
+                                  cache_pull(File.expand_path(cache),
+                                             ignore: options[:cache_ignore])
+                              else
+                                  require "autoproj/cli/ci"
+                                  cli = CI.new
+                                  cli.load_report(options[:report],
+                                                  "cache_pull_report")["packages"]
+                              end
+
```

Comment:
I'd say ... make the whole logic part of a separate helper, which you can then use in both osdeps and build (and would probably solve the complexity cop)

Also, I would recommend making the `cache` path deprecated - i.e. people would have to explicitly call cache-pull in their workflow (which is, I am assuming, what you do in your workflow)

Reply:
> Also, I would recommend making the cache path deprecated

Yeah, I actually wanted to keep the current behavior to not force _you_ to change your workflow. If that's ok for you, I will do that.